common/lib/xmodule/xmodule/modulestore/tests/test_split_modulestore.py

```diff
@@ -888,6 +889,25 @@ def test_block_generations(self, _from_json):
         version_history = modulestore().get_block_generations(second_problem.location)
         self.assertNotEqual(version_history.locator.version_guid, first_problem.location.version_guid)
 
+    def test_make_course_usage_key(self):
+        """Test that we get back the appropriate usage key for the root of a course key.
+
+        In particular, we want to make sure that it properly handles CCX courses.
+        """
+        normal_course_key = CourseKey.from_string("course-v1:edx+test_course+test_run")
+        ccx_key = CourseKey.from_string("ccx-v1:edX+test_course+test_run+ccx@1")
+
+        normal_root_key = modulestore().make_course_usage_key(normal_course_key)
+        ccx_root_key = modulestore().make_course_usage_key(ccx_key)
+
+        self.assertIsInstance(normal_root_key, BlockUsageLocator)
+        self.assertIsInstance(ccx_root_key, CCXBlockUsageLocator)
+
+        # They should both be Course modules..
+        for usage_key in [normal_root_key, ccx_root_key]:
+            self.assertEqual(usage_key.block_type, "course")
+            self.assertEqual(usage_key.name, "course")
```

Comment:
Nit: consider using ddt to test the various key types.  In the future, if we ever add another CourseKey type, we can more easily add the new case in the ddt.  Each ddt case would include both its "string" value and its "CourseKey" type.